terraform/modules/happy-tfe-okta-service-account/main.tf

```diff
@@ -0,0 +1,50 @@
+resource "aws_kms_key" "service_user" {
+  description              = local.label
+  key_usage                = "SIGN_VERIFY"
+  customer_master_key_spec = "RSA_4096"
+}
+data "aws_kms_public_key" "service_user" {
+  key_id = aws_kms_key.service_user.key_id
+}
+data "jwks_from_key" "jwks" {
+  key = data.aws_kms_public_key.service_user.public_key_pem
+  kid = aws_kms_key.service_user.key_id
+}
+locals {
+  jwks  = jsondecode(data.jwks_from_key.jwks.jwks)
+  label = "${var.service_name}-${var.app_name}-service-account"
+}
+
+module "happy_app" {
+  source = "git@github.com:chanzuckerberg/shared-infra//terraform/modules/okta-app-oauth?ref=heathj/jwks"
+
+  okta = {
+    label         = local.label
+    redirect_uris = concat(["https://oauth.${var.app_name}.si.czi.technology/oauth2/callback"], var.redirect_uris)
+    login_uri     = var.login_uri == "" ? "https://oauth.${var.app_name}.si.czi.technology" : var.login_uri
```

Comment:
do these need the env in the URI as well?

Reply:
I was thinking the service user would be shared between happy envs (prod,staging,rdev) so that we could share the key across all workspaces. But maybe we should make one per env instead so that we can differentiate the clients so that, for instance, rdev can't edit the config of prod or something

Reply:
Yeah I'm in favor of one per env